mypy/main.py

```diff
@@ -62,13 +62,27 @@ def main(script_path: Optional[str], args: Optional[List[str]] = None) -> None:
         args = sys.argv[1:]
     sources, options = process_options(args)
     serious = False
+    errors = False
+
+    def flush_errors(a: List[str]) -> None:
+        nonlocal errors
+        if a:
+            errors = True
+        f = sys.stderr if serious else sys.stdout
+        try:
+            for m in a:
+                f.write(m + '\n')
+        except BrokenPipeError:
+            pass
+
     try:
-        res = type_check_only(sources, bin_dir, options)
+        res = type_check_only(sources, bin_dir, options, flush_errors)
```

Comment:
I'd prefer if the main body of test cases (`mypy/test/testscheck.py`) would use streaming errors (without testing that flushing happens as expected, i.e. we'd still only test that the sequence of errors in the output is as expected). Not sure if this is feasible without many changes to test cases. If not, at least it would be good to do a one-off manual check that streaming produces the same errors as not streaming in test cases.